data_types/src/lib.rs

```diff
@@ -63,6 +63,16 @@ impl TryFrom<i32> for CompactionLevel {
     }
 }
 
+impl Display for CompactionLevel {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Initial => write!(f, "0"),
+            Self::FileNonOverlapped => write!(f, "1"),
+            Self::Final => write!(f, "2"),
+        }
```

Comment:
```suggestion
        write!(f, "{}", self as i16)
```

This works because of:

```rust
#[repr(i16)]
pub enum CompactionLevel {
...
```

Reply:
Interestingly, I got the error:
```
error[E0606]: casting `&CompactionLevel` as `i16` is invalid
  --> data_types/src/lib.rs:68:25
   |
68 |         write!(f, "{}", self as i16)
   |                         ^^^^^^^^^^^
   |
   = help: cast through a raw pointer first
```

Reply:
Sorry, I've missed a de-ref `*`:

`write!(f, "{}", *self as i16)`